mne_bids/utils.py

```diff
@@ -349,6 +353,55 @@ def _extract_landmarks(dig):
     return coords
 
 
+def _find_best_candidates(params, candidate_list):
+    """Return the best candidate, based on the number of param matches.
+
+    Assign each candidate a score, based on how many entities are shared with
+    the ones supplied in the `params` parameter. The candidate with the highest
+    score wins. Candidates with entities that conflict with the supplied
+    `params` are disqualified.
+
+    Parameters
+    ----------
+    params : dict
+        The entities that the candidate should match.
+    candidate_list : list of str
+        A list of candidate filenames.
+
+    Returns
+    -------
+    best_candidates : list of str
+        A list of all the candidate filenames that are tied for first place.
+        Hopefully, the list will have a length of one.
+    """
+    params = {key: value for key, value in params.items() if value is not None}
+
+    best_candidates = []
+    best_score = 0
```

Comment:
The PR looks great. I have only one minor nitpicky comment. It is about this variable. Perhaps it could be harmonized with `n_matches` to be something like `n_matches_best` or something like that? I was wondering for some time how the two were different

Reply:
done!